src/utils.rs

```diff
@@ -0,0 +1,91 @@
+use sysinfo::{Pid, ProcessExt, SystemExt};
+
+#[cfg(test)]
+const SKIP_ARGS: usize = 1;
+#[cfg(not(test))]
+const SKIP_ARGS: usize = 2;
+
+fn guess_filename_extension_from_args(args: &[String]) -> Option<String> {
+    let mut saw_dash_dash = false;
+    // skip "git blame" (but during testing only "cargo", not "test" as well)
+    for x in args.iter().skip(SKIP_ARGS) {
+        if x == "--" {
+            saw_dash_dash = true;
+            continue;
+        }
+        if saw_dash_dash || !x.starts_with('-') {
+            return x.split('.').last().map(str::to_owned);
+        }
+    }
+
+    None
+}
```

Comment:
I wonder if there is a crate dedicated to parsing arbitrary command lines known only at run time?

Reply:
Since this is more a command line ignorer than parser it is a rather simple function, see `skip_uninteresting_args`.